# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  319b3df682f1f1e116be54aaf1222c462f874d64
NNN26 - Add data: and blob: to CSP of compose window, see bug 1964799.

This is necessary for WebExtension experiments where 
"content_security_policy": "default-src 'self'; script-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; worker-src 'self' blob:"
in the manifest doesn't work.

diff --git a/mail/components/compose/content/messengercompose.xhtml b/mail/components/compose/content/messengercompose.xhtml
--- a/mail/components/compose/content/messengercompose.xhtml
+++ b/mail/components/compose/content/messengercompose.xhtml
@@ -42,9 +42,9 @@
 <head>
   <title>&msgComposeWindow.title;</title>
   <meta
     http-equiv="Content-Security-Policy"
-    content="default-src chrome:; script-src chrome: resource: 'unsafe-inline'; style-src chrome: 'unsafe-inline'; img-src chrome: moz-icon:; worker-src resource:; connect-src http: https:"
+    content="default-src chrome:; script-src chrome: resource: 'unsafe-inline'; style-src chrome: 'unsafe-inline'; img-src chrome: moz-icon: data:; worker-src resource: blob:; connect-src http: https:"
   />
   <link
     rel="stylesheet"
     href="chrome://messenger/skin/messengercompose/messengercompose.css"
